Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java

```diff
@@ -217,6 +219,129 @@ public void driveSlidesWithController() {
+        // joystick control
+        slideTargetPosition += (int) (-gamepad2.left_stick_y * 25);
+
+        // cone stack positions - increase position by one if dpad up is just pressed
+        if (gamepad2.dpad_up && previousConeStack == currentConeStack) {
+            currentConeStack++;
+
+            // don't let dpad control take slides above highest cone height in cone stack
+            if (currentConeStack >= 4) {
+                currentConeStack = 4;
+            }
+
+            // set slide target position to the chosen cone stack height
+            switch (currentConeStack) {
+                case 1:
+                    slideTargetPosition = Constants.SLIDE_STACK_ONE;
+                    break;
+                case 2:
+                    slideTargetPosition = Constants.SLIDE_STACK_TWO;
+                    break;
+                case 3:
+                    slideTargetPosition = Constants.SLIDE_STACK_THREE;
+                    break;
+                case 4:
+                    slideTargetPosition = Constants.SLIDE_STACK_FOUR;
+                    break;
+            }
+
+            // cone stack positions - decrease position by one if dpad down is just pressed
+        } else if (gamepad2.dpad_down && previousConeStack == currentConeStack) {
+            currentConeStack--;
+
+            // don't let dpad control take slides below ground height
+            if (currentConeStack <= 0) {
+                currentConeStack = 0;
+            }
+
+            // set slide target position to the chosen cone stack height
+            switch (currentConeStack) {
+                case 0:
+                    slideTargetPosition = Constants.SLIDE_BOTTOM;
+                    break;
+                case 1:
+                    slideTargetPosition = Constants.SLIDE_STACK_ONE;
+                    break;
+                case 2:
+                    slideTargetPosition = Constants.SLIDE_STACK_TWO;
+                    break;
+                case 3:
+                    slideTargetPosition = Constants.SLIDE_STACK_THREE;
+                    break;
+            }
+        }
+
+        // junction positions - increase position by one if right bumper is just pressed
+        if (gamepad2.right_bumper && previousJunction == currentJunction) {
+            currentJunction++;
+
+            // don't let bumper control take slides above high junction height
+            if (currentJunction >= 3) {
+                currentJunction = 3;
+            }
+
+            // set slide target position to the chosen junction height
+            switch (currentJunction) {
+                case 1:
+                    slideTargetPosition = Constants.SLIDE_LOW;
+                    break;
+                case 2:
+                    slideTargetPosition = Constants.SLIDE_MEDIUM;
+                    break;
+                case 3:
+                    slideTargetPosition = Constants.SLIDE_HIGH;
+                    break;
+            }
```

Comment:
This could be broken out into a helper function that passes in `currentJunction` and returns the `slideTargetPosition`.

Reply:
And here